app/pages/resource/ResourcePage.js

```diff
@@ -107,32 +108,63 @@ class UnconnectedResourcePage extends Component {
     this.setState(() => ({ isOpen: true, photoIndex }));
   };
 
-  orderImages = (images) => {
+  getOrderdNormalImages = (images) => {
     return [].concat(
       images.filter(image => image.type === 'main'),
-      images.filter(image => image.type !== 'main'),
+      images.filter(image => (image.type !== 'main' && image.type !== 'panoroma')),
     );
   };
 
+  getPanoromaImages = (images) => {
+    const panoromaImages = images.filter(image => image.type === 'panoroma');
+    return panoromaImages.map(image => (
+      <Pannellum
+        autoLoad
+        height="98%"
+        hfov={100}
+        image={image.url}
+        imageCaption={image.caption}
+        imageType="panoroma"
+        key={image.url}
+        pitch={10}
+        width="100%"
+        yaw={180}
+      />
+    ));
+  }
+
   renderImage = (image, index, { mainImageMobileVisibility = false }) => {
     const isMainImage = image.type === 'main';
+    const isPanoromaImage = this.imageIsPanoroma(image);
     const className = classNames('app-ResourceInfo__image-wrapper', {
       'app-ResourceInfo__image-wrapper--main-image': isMainImage,
       'app-ResourceInfo__image-wrapper--mobile-main-image':
         isMainImage && mainImageMobileVisibility,
     });
 
+    // Panoroma/360 are not normal images, they are react component. Thus, build the
+    // image props for panoroma images to match with rest of the images.
+    let panoromaImage = {};
+    if (isPanoromaImage) {
+      const panoromaImgProps = image.props;
+      panoromaImage = {
+        url: panoromaImgProps.image,
+        image: panoromaImgProps.image,
+        caption: panoromaImgProps.imageCaption,
+      };
+    }
+
     return (
-      <div className={className} key={image.url}>
+      <div className={className} key={isPanoromaImage ? panoromaImage.url : image.url}>
         <button
           className="app-ResourceInfo__image-button"
           onClick={() => this.handleImageClick(index)}
           type="button"
         >
           <img
-            alt={image.caption}
+            alt={isPanoromaImage ? panoromaImage.caption : image.caption}
             className="app-ResourceInfo__image"
-            src={this.getImageThumbnailUrl(image)}
+            src={isPanoromaImage ? this.getImageThumbnailUrl(panoromaImage) : this.getImageThumbnailUrl(image)}
```

Comment:
Instead of doing 3 if/else instead of the jsx can you make one before the jsx

Reply:
I think its better to fix the code style in separate PR. I will fix all the style related changes there.